### PR TITLE
Return proper error message while stream gets created with mismatching configs

### PR DESCRIPTION
**Pull request by prabhakar1983**
# stream-registry PR

1. Changed all the service layer exceptions to CheckedExceptions
2. Removed the Business objects from Custom Exceptions
3. Refactored the exception chaining so that proper exceptions are bubbled up
4. Standardized the ErrorMessage sent back to Client

# PR Checklist Forms

- [x] Reviewer assigned
- [x] PR assigned (presumably to submitter)
- [x] Labels added (enhancement, bug, documentation)

### CODE REVIEW

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/exceptions/BaseStreamException.java [hunk 1]
```diff
@@ -27,7 +27,8 @@ public BaseStreamException(Throwable cause, String streamName) {
         this.streamName = streamName;
     }
 
-    public BaseStreamException(String streamName) {
+    public BaseStreamException(String streamName, String message) {
```
Comment (prabhakar1983) [inline core/src/main/java/com/homeaway/streamplatform/streamregistry/exceptions/BaseStreamException.java]: earlier it was just accepting streamName which was calling super(message).

When the exception was handled, the e.getMessage was null.

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/exceptions/StreamDeletionException.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
 public class StreamDeletionException extends BaseStreamException {
     private static final long serialVersionUID = -5793888606557229357L;
 
-    public StreamDeletionException(String streamName) {
-        super(streamName);
```
Comment (neoword) [inline core/src/main/java/com/homeaway/streamplatform/streamregistry/exceptions/StreamDeletionException.java]: This should not be `streamName` this should be `msg` and we should fix everything to have the appropriate _*msg*_!!! 😄

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/exceptions/StreamNotFoundException.java [hunk 3]
```diff
@@ -19,6 +19,10 @@
     private static final long serialVersionUID = -2171171630296498063L;
 
     public StreamNotFoundException(String streamName) {
-        super(streamName);
```
Comment (neoword) [inline core/src/main/java/com/homeaway/streamplatform/streamregistry/exceptions/StreamNotFoundException.java]: Same here as above.

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/db/dao/impl/KafkaManagerImpl.java [hunk 4]
```diff
@@ -128,7 +128,7 @@ void updateTopics(ZkUtils zkUtils, List<String> topicsToUpdate, Map<String, Stri
             // TODO Alternatively we can add a forceSync=true flag, ignoring any user provided info, and only updating SR with the underlying settings
             //      We should probably do forceSync=true anyway, as it provides a simple way to keep things in sync (#114)
             if(isNewStream) {
-                throw new StreamCreationException(String.format("Error: Input configs=%s and actual configs=%s are not same for topic=%s", topicConfigMap, actualTopicConfig, topic));
+                throw new StreamCreationException(topic, String.format("Error: Input configs=%s and actual configs=%s are not same for topic=%s", topicConfigMap, actualTopicConfig, topic));
```
Comment (neoword) [inline core/src/main/java/com/homeaway/streamplatform/streamregistry/db/dao/impl/KafkaManagerImpl.java]: please include topic in msg

**Comment by prabhakar1983** [comment]
@neoword 
1) converted all the Custom RuntimeExceptions to Checked Exceptions so the Resource class should handle them with proper error messages
2) Add documentation for all the Interfaces. 
3) Made sure the methods on the Custom Exceptions are appropriate.

File: core/src/main/java/com/homeaway/streamplatform/streamregistry/db/dao/impl/ConsumerDaoImpl.java [hunk 5]
```diff
@@ -151,42 +155,53 @@ public void delete(String streamName, String actorName) {
...
-                if (consumer.getActor().getName().equals(consumerName))
-                    return Optional.of(AvroToJsonDTO.getJsonConsumer(consumer));
+            if (streamValue.get().getConsumers() != null) {
+                for (com.homeaway.digitalplatform.streamregistry.Consumer consumer : streamValue.get().getConsumers()) {
+                    if (consumer.getActor().getName().equals(consumerName))
+                        return Optional.of(AvroToJsonDTO.getJsonConsumer(consumer));
+                }
             }
+        } else {
+            throw new StreamNotFoundException(String.format("StreamName=%s not found. Please create the Stream before getting a Consumer", streamName));
```
Comment (neoword) [inline core/src/main/java/com/homeaway/streamplatform/streamregistry/db/dao/impl/ConsumerDaoImpl.java]: can we invert this if/else statement?
```
if (!streamValue.isPresent()) {
  throw new StreamNotFoundException(String.format("StreamName=%s not found. Please create the Stream before getting a Consumer", streamName));
}
```

Comment (prabhakar1983) [reply core/src/main/java/com/homeaway/streamplatform/streamregistry/db/dao/impl/ConsumerDaoImpl.java]: @neoword 
fixed it.

Threw the StreamNotFoundException right away at the top of the methods

```registerConsumer
getConsumer
getAllConsumers
deleteConsumers```

```registerProducer
getProducer
getAllProducers
deleteProducer```

**Pull request merged**